eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Management/src/Providers/OperationMethodProviders/ResourceOperationMethodProvider.cs

```diff
@@ -226,29 +265,29 @@ private IReadOnlyList<MethodBodyStatement> BuildFakeLroHandling(
 
         private IReadOnlyList<MethodBodyStatement> BuildLroHandling(
             VariableExpression messageVariable,
-            VariableExpression responseVariable,
+            ScopedApi<Response> responseVariable,
             ParameterProvider cancellationTokenParameter)
         {
             var statements = new List<MethodBodyStatement>();
 
             var finalStateVia = _serviceMethod.GetOperationFinalStateVia();
 
-            var armOperationType = _isGeneric
-                ? ManagementClientGenerator.Instance.OutputLibrary.GenericArmOperation.Type
-                    .MakeGenericType([_resource.Type])
+            var armOperationType = _returnBodyType != null
+                ? ManagementClientGenerator.Instance.OutputLibrary.ArmOperationOfT.Type
+                    .MakeGenericType([_returnBodyType])
                 : ManagementClientGenerator.Instance.OutputLibrary.ArmOperation.Type;
 
             ValueExpression[] armOperationArguments = [
                 _clientDiagnosticsField,
-                This.Property("Pipeline"),
+                This.As<ArmResource>().Pipeline(),
                 messageVariable.Property("Request"),
                 responseVariable,
                 Static(typeof(OperationFinalStateVia)).Property(finalStateVia.ToString())
             ];
 
-            var operationInstanceArguments = _isGeneric
+            var operationInstanceArguments = _resourceClient != null
```

Comment:
this change of condition fixes the issue https://github.com/Azure/azure-sdk-for-net/issues/51177